### PR TITLE
Create ThriftBufferResult and ThriftSerializedPage

### PR DESCRIPTION
**Pull request by viczhang861**
Make SerializedPage independent of thrift serialization.

The goal is to move SerializedPage and related classes to spi module. This PR makes migration easier.
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftSerializedPage.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    private final int positionCount;
+    private final int uncompressedSizeInBytes;
+    private final byte pageCodecMarkers;
+
+    public ThriftSerializedPage(SerializedPage serializedPage)
+    {
+        this(serializedPage.getSlice(), serializedPage.getPageCodecMarkers(), serializedPage.getPositionCount(), serializedPage.getUncompressedSizeInBytes());
+    }
+
+    public ThriftSerializedPage(Slice slice, byte pageCodecMarkers, int positionCount, int uncompressedSizeInBytes)
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftSerializedPage.java]: private

File: presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        return new ThriftBufferResult(
+                bufferResult.getTaskInstanceId(),
+                bufferResult.getToken(),
+                bufferResult.getNextToken(),
+                bufferResult.isBufferComplete(),
+                thriftSerializedPages);
+    }
+
+    @ThriftConstructor
+    public ThriftBufferResult(String taskInstanceId, long token, long nextToken, boolean bufferComplete, List<ThriftSerializedPage> thriftSerializedPages)
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java]: Add javadoc to denote there should be no explicit call on this method.

Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java]: nit: one line per parameter

File: presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+@ThriftStruct
+public class ThriftBufferResult
+{
+    private final String taskInstanceId;
+    private final long token;
+    private final long nextToken;
+    private final boolean bufferComplete;
+    private final List<ThriftSerializedPage> thriftSerializedPages;
+
+    public static ThriftBufferResult createThriftBufferResult(BufferResult bufferResult)
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/execution/buffer/ThriftBufferResult.java]: nit: what about `from` or `fromBufferResult`?  This seems to be the convention for factory method name in Presto, for example, see `RowType::from`, `FieldId::from`, etc :)

**Comment by viczhang861** [comment]
@highker Unit test for serde added.  I have no idea why Travis build is not complete given all tests actually pass.

**Pull request merged**